torch/csrc/cuda/MemPool.cpp

```diff
@@ -11,9 +12,16 @@ using shared_ptr_class_ = py::class_<T, std::shared_ptr<T>>;
 void THCPMemPool_init(PyObject* module) {
   auto torch_C_m = py::handle(module).cast<py::module>();
   shared_ptr_class_<::c10::cuda::MemPool>(torch_C_m, "_MemPool")
-      .def(py::init<c10::cuda::CUDACachingAllocator::CUDAAllocator*, bool>())
+      .def(
+          py::init([](c10::cuda::CUDACachingAllocator::CUDAAllocator* allocator,
+                      bool is_user_created) {
+            torch::utils::device_lazy_init(at::kCUDA);
```

Comment:
New. Seems ok though.

Reply:
Yes. That's because I have a device parameter in the MemPool object now.